sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala

```diff
@@ -230,6 +236,15 @@ trait FileFormat {
   }
 
   /**
+   * Return a `SerializablePathFilter` class to filter qualified files for this format.
+   */
+  def getPathFilter(): SerializablePathFilter = {
```

Comment:
Shall we add either the data source options map or the Hadoop conf as an argument of this method?

For example, the Avro data source may filter out all input files whose file names don't end with ".avro" if Hadoop conf "avro.mapred.ignore.inputs.without.extension" is set to true. This is consistent with default behavior of `AvroInputFormat`.